### DISCUSSION TITLE
Is the following construction vulnerable to XSS?

### DISCUSSION THREAD

**Discussion by bananabr** [jquery/jquery#4899] [Q&A] [answered]
I am evaluating a static analysis tool and the following construction was identified in one of my projects as vulnerable to XSS. I couldn't come up with a value from location.hash that would build a valid exploit so I think it's a false positive. Could you please confirm?

```js
$(`#foo ${location.hash}`);
```

Thanks!

**Comment by mgol** [comment, marked as answer]
A tool probably knew that passing `location.hash` directly to `jQuery` may allow you to construct an element due to the ambiguity of the `jQuery` function. However, jQuery >=1.9 requires input resulting in creating new elements to start with `<`, possibly preceded by some whitespace. Therefore, I don't see a security risk in your specific input.

Note, though, that it may contain an invalid selector that will make the call crash so it's a good idea to escape such an input anyway.

**Comment by xss-pariah** [comment]
with code in general, i feel like to say, you may not notice nor see possible vulnerabilities within the areas in question or topic, however. i will say to my experience with tech in general, simply because we don't think it can be done now, doesn't mean it cant be done period. its a matter of knowledge, and the minute we stop thinking we can exploit, or find a vuln, (much like with xss) then we stop improving as developers and engineers as a whole. so security isnt always as secure as we want it to be, its as secure as we KNOW it to be within the best of your current knowledge. i am a very fresh player on the board still by far in this field. but i am very passionate about the idea of xss, sql, and many other vulnerability-based (not attacks) but educational exploration with script exploitation to better assist those who are very passionate about the development side to coding. i find that exploitation and other errors we come across serves a few purposes in this instance. one, it helps the one who coded it to build better around that vulnerability , and two it helps us learn from it. So, just because in this context you may not see a xss possibility, doesn't mean that in  a full script with more to it than as shown above there wont be a vulnerability at all. 
now whether its a false positive, the guy above is definitely accurate.
